### PR TITLE
Part of FUSETOOLS2-1579: Add simple language autocompletion (basic)

### PR DESCRIPTION
**Pull request by Delawen**
Add some simple language placeholders that can be useful. It does not check the context or the route to extract "dynamic" values like headers or exchangeProperties.

The values provided with this auto-completion are:

`${camelId}`, `${exchange}`, `${exchangeId}`, `${id}`,`${messageTimestamp}`, `${body}`, `${bodyOneLine}`, `${prettyBody}`, `${headers}`, `${exception}`, `${exception.message}`, `${exception.stacktrace}`, `${routeId}`, `${stepId}`, `${threadId}`, `${threadName}`, `${hostname}`, `${null}`, `${messageHistory}`

It also refactor common parts of the code with the kubernetes auto-completion of secrets and config maps.

[Screencast from 2023-10-18 10-47-41.webm](https://github.com/camel-tooling/camel-language-server/assets/726590/d5141f10-df6c-485e-be05-d509f3785446)

### CODE REVIEW

File: src/test/java/com/github/cameltooling/lsp/internal/completion/CamelKubernetesServicesCompletionTest.java [hunk 1]
```diff
@@ -193,6 +193,137 @@ void testKubernetesAutoCompletionPlaceholdersNotClosed() throws Exception {
 		assertThat(completions.get(1).getInsertText()).isEqualTo("}}something{{configmap:myMap/myKey2}}");
 	}
 
+	@Test
+	void testSimpleLanguage() throws Exception {
```
Comment (apupier) [inline src/test/java/com/github/cameltooling/lsp/internal/completion/CamelKubernetesServicesCompletionTest.java:197]: I think the test for variables deserves a specific Test class as it is not related to Kubernetes only.

File: src/test/java/com/github/cameltooling/lsp/internal/completion/CamelKubernetesServicesCompletionTest.java [hunk 2]
```diff
@@ -193,6 +193,137 @@ void testKubernetesAutoCompletionPlaceholdersNotClosed() throws Exception {
...
+		String camelUri = "pgevent:host:999/database/channel?user=something${";
```
[the remark this answers is not in the document]
Comment (Delawen) [reply src/test/java/com/github/cameltooling/lsp/internal/completion/CamelKubernetesServicesCompletionTest.java:198]: Oh, I was quite sure that you could add simple language  everywhere. Now I am doubting :)

I'll check and make sure.

File: src/test/java/com/github/cameltooling/lsp/internal/completion/CamelKubernetesServicesCompletionTest.java (hunk 1 above)
Comment (Delawen) [reply src/test/java/com/github/cameltooling/lsp/internal/completion/CamelKubernetesServicesCompletionTest.java:197]: You are right, I forgot about the name of the class

**Comment by apupier** [comment]
status reported in specific JIRA https://issues.redhat.com/browse/FUSETOOLS2-2268

**Pull request closed without merging**